File: program/SESI5/2/prima2.pas
var
   N, cur, i : longint;
   count : longint;
   found : boolean;
begin
   readln(N);

   count := 0; // banyaknya prima yang sudah ditemukan
   cur := 2; // nilai yang sedang diperiksa keprimaannya
   while (count < N) do begin
      found := FALSE;
      i := 2;
      while ((i < cur) and not found) do begin
         if (cur mod i = 0) then begin
            found := TRUE;
         end;
         i := i + 1;
      end;

      if (not found) then begin
         // ditemukan prima! 
         // cetak dan tambahkan prima yg sudah ditemukan
         writeln(cur);
         count := count + 1;
      end;

      // entah ini prima atau bukan, lanjut untuk 
      // memeriksa bilangan berikutnya
      cur := cur + 1;
   end;
end.

File: topik/pemrograman-dasar-cpp/07-perulangan-lanjut/asset/prima_2.cpp
#include <cstdio>

int main() {
  int N;
  scanf("%d", &N);

  int count = 0; // Banyaknya prima yang sudah ditemukan
  int cur = 2; // nilai yang akan diperiksa keprimaannya

  while (count < N) {
    bool prima = true;
    for (int i = 2; i <= cur-1; i++) {
      if (cur % i == 0) {
        prima = false;
        break;
      }
    }

    if (prima) {
      // Ditemukan prima!
      // Cetak dan tambahkan prima yg sudah ditemukan
      printf("%d\n", cur);
      count++;
    }

    // Entah ini prima atau bukan, lanjut untuk
    // memeriksa bilangan berikutnya
    cur++;
  }
  // Keluar dari while, dipastikan count = N
}
